### PR TITLE
Use docker-compose for local development

### PR DESCRIPTION
**Pull request by marcmillien**
# Context
Setup `docker-compose` environment for local development.

# Why
So local usage and all dependencies installation become as simple as running: `docker-compose up`.

This command will make docker run `ruby2.4` on script `scripts/web` that will install all requirements then start `mailcatcher`, that will be available on port 8080 (web) et 8025 (smtp).

Test suite can be run by using `docker-compose run --rm ruby2.4 scripts/test`, that will install all requirements then run `rake`.

# Additionnal informations

## /deps
The setup use `deps` as directory to keep dependencies docker execution after docker execution (so we do not reinstall `nodejs`, `phantomjs` and `gem` dependencies at each `docker-compose up`).

## Ruby version
Remove `1.9.3` from test:
- because it is no more a maintained version
- due to minimum requirement for `selenium-webdriver` 

## Tests
Sounds like tests are failing due to some race condition, so I forced the navigation after delivering mail.

## Dependencies
Finally this PR fix versions for:
- `selenium-webdriver` due to the fact that phantomjs is no more supported by selenium webdriver since version `3.8`
- `compass` due to the error: `NoMethodError: undefined method 'has?' for Sass::Util:Module`

## References
PR opened to upstream: https://github.com/sj26/mailcatcher/pull/374

### CODE REVIEW

**Comment by KillianKemps** [comment]
Ma partie préférée : "PR opened to upstream: sj26#374"

**Comment by KillianKemps** [comment]
Typo dans la description : 
"The setup use deps *as*"

File: CONTRIBUTE.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# How to contribute
+
+## Using docker + docker-compose
+
+Run `docker-compose up`, it will install all requirements and start mailcatcher that will be available on port 8080 for the web interface and 8025 for stmp.
```
Comment (KillianKemps) [inline CONTRIBUTE.md]: typo : `stmp`-> `smtp`

**Review by KillianKemps — APPROVED** [review]
Quelques typos mais sinon :+1:

File: CONTRIBUTE.md (hunk 1 above)
Comment (marcmillien) [reply CONTRIBUTE.md]: Je fixe ça :).

**Comment by marcmillien** [comment]
> Ma partie préférée : "PR opened to upstream: sj26#374"

Vu le nombre de PR ouvertes sur l'upstream pas sûr que ce soit mergé, mais on verra :).

**Pull request merged**